src/shogun/base/SGObject.cpp

```diff
@@ -1135,6 +1135,12 @@ void CSGObject::push_back(CDynamicObjectArray* array, CSGObject* value)
 	array->push_back(value);
 }
 
+CSGObject* CSGObject::get(CDynamicObjectArray* array, index_t index)
+{
+	ASSERT(array);
+	return array->get_element(index);
```

Comment:
Is this throwing any errors to the user if the supplied index is out-of-bound/negative?

Reply:
yes, `CDynamicObjectArray` takes care of that. We might want to catch the expection and make a pretty printed version since this is user facing though. Will add that